kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -259,6 +259,7 @@ class ResidualBasedNewtonRaphsonStrategy
      
     virtual ~ResidualBasedNewtonRaphsonStrategy()
     {
+        Clear();
```

Comment:
@jcotela this is the test case I used.

https://gigamove.rz.rwth-aachen.de/d/id/es4QsU52KUvxEX?11&id=es4QsU52KUvxEX

Here is the output before the commit:

#0  0x00007fffe2fbde43 in ML_Ifpack_Destroy ()
   from /home/mike/Software/kratos/trilinos/lib/libml.so.11
#1  0x00007fffe2ed371f in ML_Smoother_Clean ()
   from /home/mike/Software/kratos/trilinos/lib/libml.so.11
#2  0x00007fffe2e7df8b in ML_Destroy ()
   from /home/mike/Software/kratos/trilinos/lib/libml.so.11
#3  0x00007fffe2fd8041 in ML_Epetra::MultiLevelPreconditioner::DestroyPreconditioner() () from /home/mike/Software/kratos/trilinos/lib/libml.so.11
#4  0x00007fffe2ff5075 in ML_Epetra::MultiLevelPreconditioner::~MultiLevelPreconditioner() () from /home/mike/Software/kratos/trilinos/lib/libml.so.11
#5  0x00007ffff40d7dfe in boost::detail::sp_counted_base::release() ()
   from /home/mike/Software/kratos/kratos/libs/libboost_python.so.1.59.0
#6  0x00007fffe4e42822 in ~shared_count (this=<optimized out>, 
    __in_chrg=<optimized out>)
    at /home/mike/Software/kratos/boost/include/boost/smart_ptr/detail/shared_count.hpp:467
#7  ~shared_ptr (this=<optimized out>, __in_chrg=<optimized out>)
    at /home/mike/Software/kratos/boost/include/boost/smart_ptr/shared_ptr.hpp:330
#8  reset (this=0x1545018)
    at /home/mike/Software/kratos/boost/include/boost/smart_ptr/shared_ptr.hpp:653
#9  ResetPreconditioner (this=0x1544eb0)